### DISCUSSION TITLE
[Feedback] Having trouble with the temporarily-recommended process of deploying my own OptimismMintableERC20Factory

### DISCUSSION THREAD

**Discussion by JayWelsh** [ethereum-optimism/developers#147] [Feedback]
### Feedback

This is in reference to the following topic: https://github.com/ethereum-optimism/developers/discussions/143

I wanted to provide feedback as I've had trouble with performing this temporarily-recommended process, and perhaps this feedback will show that it's worth organising official L2 predeployments with the new `OptimismMintableERC20Factory` which supports custom decimal places. Or alternatively, maybe I can understand what I am doing wrong with my attempt to deploy the factory.

Firstly, I am working on Base Network (I believe this is the part that is complicating the situation).

This is the process I am following:

1. Cloned `https://github.com/ethereum-optimism/optimism`
2. Opened the directory, `cd` into `optimism/packages/contracts-bedrock`
3. `pnpm install`
4. `forge install`
5. Set `ETH_RPC_URL`, `DEPLOYMENT_ADDRESS_PRIVATE_KEY`, `ETHERSCAN_API_KEY` & `DEPLOYMENT_CONTEXT` env vars
6. Ran `forge script -vvv scripts/Deploy.s.sol:Deploy --sig 'deployOptimismMintableERC20Factory()' --rpc-url $ETH_RPC_URL --broadcast --private-key $DEPLOYMENT_ADDRESS_PRIVATE_KEY --verify`

This process does indeed work just fine for some networks, the networks which it works for are any networks contained in the following folder (i.e. it's fine if I set the `DEPLOYMENT_CONTEXT` to any of these networks): https://github.com/ethereum-optimism/optimism/tree/c04cefe0258a3a13266fe7f3df830f392f4a14dc/packages/contracts-bedrock/deployments

However, the issue is that I need to be able to deploy this factory to `base-sepolia` & `base`, when I use either of these as the `DEPLOYMENT_CONTEXT` values, the process fails. It seems that I would be required to deploy the whole suite of contracts to `base` & `base-sepolia` in order to make the command in step 6 above work. Keeping this in mind, I don't think this is a viable approach because it seems to me (unless I am misunderstanding something), that I'd be deploying many more contracts than I actually should be, and this newly deployed entire suite of contracts would *all* be deviating from the current official deployments by Optimism / Base, as opposed to the only deviation being in the `OptimismMintableERC20Factory` contract.

The Base team expressed that they would make a deployment of these contracts themselves once they are part of an official OP Stack release.

In summary, it seems fairly viable to take the approach of deploying a new `OptimismMintableERC20Factory` contract, if that is the only deviation from the current official deployments by Optimism / Base. But it seems like an unworkable scenario if the entire suite of contracts needs to be redeployed (since it's many complex contracts that I wouldn't be able to handle the "maintenance" of, since many are complex proxies that I probably shouldn't be taking "ownership" of).

I'm open to any feedback or thoughts regarding this, it's very possible that I'm doing something wrong or misunderstanding/overestimating the risks involved in deploying the whole suite of contracts.

Thanks a lot in advance! :)

**Comment by sbvegan** [comment]
Hey @JayWelsh, thank you for taking the time to provide this feedback. Some comments that I've received from a couple engineers:

- Developers can directly deploy an `OptimismMintableERC20` contract without needing the factory to deploy it
- The factory contract is considered, "extraneous utility" because nothing in the rest of the L1 system points to it

Your feedback is valid. I don't think it will be prioritized in the near term because there are other avenues developers can take to be unblocked.

**Reply by JayWelsh** [reply]
Thanks for this! You have clarified my points of confusion. I appreciate the help and will follow this guidance accordingly! 👍
